apps/prairielearn/src/pages/studentAssessmentInstanceExam/studentAssessmentInstanceExam.sql

```diff
@@ -27,7 +39,22 @@ SELECT
   (lag(aq.effective_advance_score_perc) OVER w) AS prev_advance_score_perc,
   'Question ' || (lag(qo.question_number) OVER w) AS prev_title,
   (lag(qo.sequence_locked) OVER w) AS prev_sequence_locked,
-  iqnag.*
+  iqnag.*,
+  (
+    SELECT
+      bool_or(aqrp.can_view)
+    FROM
+      assessment_questions AS aq
+      JOIN assessment_question_role_permissions AS aqrp ON aqrp.assessment_question_id = aq.id
+    WHERE
+      aq.id = iq.assessment_question_id
+      AND aqrp.group_role_id IN (
+        SELECT
+          group_role_id
+        FROM
+          group_role_ids
+      )
+  ) AS can_view
```

Comment:
Just to raise a general design question here, I see that we allow users to have multiple roles within the same group. Is this in fact desired? Lots of the code would be a lot simpler if each user only had a single role in any given group.

Reply:
Yes, this is desired. Since users can easily leave and re-join groups (which often happens if students drop classes, can't attend class on particular days, etc.), we decided to allow users to take on multiple roles so that role requirements for any assessment can be met.

Reply:
An example with POGIL roles: assessments could require a Manager, Recorder, and a Reflector to complete an assessment, with functionality specifically tailored towards each role (ex. only the Manager can assign roles, only the Recorder can answer all curriculum-related questions, and only the Reflector can fill out the groupwork-related survey at the end).

If the usual Reflector is out sick or cannot make it to class for some reason, it makes more sense for us to allow one of the other two group members to take on the Reflector role so that students can still get the most out of role-based group assessments even if their group size is unexpectedly compromised.